Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -43,9 +43,20 @@ class LoginsHelper: BrowserHelper {
     }
 
     func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        let frameInfo = message.frameInfo
+
+        // We don't currently inject the helper into iframes.
+        // Don't listen for messages from iframes, either.
+        if !frameInfo.mainFrame {
```

Comment:
Interesting. Looking at bug 1183257 again, it would have actually fixed this this vulnerability since it [adds the same `mainFrame` and `request` checks](https://github.com/mozilla/firefox-ios/pull/871/files#diff-515805a444701e1ff38e4df3e5ea40a9R52).

It looks like that PR checks the `mainFrame` for "request" messages, but allows any frame to store credentials. This seems safe to me since we don't need to worry about sending credentials to the wrong frame. The "submit" messages use the frame's URL, which I don't think can be spoofed.

That means iframes can submit form data, but not actually be autofilled. I'm not sure how useful that is, but apparently it's at least enough to fix that bug.